### ISSUE TITLE
v1: gogoproto annotations prevent valid reservation request generation

### ISSUE THREAD

**Issue by ajwerner** [mesos/mesos-go#340]
In mesos 1.4 the agent API enforces the comment on ReservationInfo that
```
// NOTE: This field must not be set for `Resource.reservation`.
```
https://github.com/mesos/mesos-go/blob/master/api/v1/lib/mesos.proto#L1253
but because of the 
```
optional Type type = 4 [(gogoproto.nullable) = false];
```
it is impossible to omit `type`. 

so we end up getting back a 400 response from the agent:
```
Header:  map[Date:[Wed, 10 Jan 2018 16:52:46 GMT] Content-Length:[117] Content-Type:[text/plain; charset=utf-8]]
Status:  400 Bad Request
Body:  Resource 'mem:30000' is invalid: 'Resource.ReservationInfo.type' must not be set for the 'Resource.reservation' field
```
We've verified that setting nullable to true, regenerating the protos, and setting type to nil resolves the issues.

**Comment by ajwerner** [comment]
this appears to be addressed by [https://github.com/mesos/mesos-go/pull/335](this pull request)

**Comment by ajwerner** [comment]
#335

**Comment by ajwerner** [comment]
What's the ETA #335? If it's more than a couple of weeks, would you be willing to accept a pull request with the single line change?

**Comment by jdef** [comment]
Thanks for this, yes willing to accept a 1 line change. I'm holding off on
1.5.x protos until Mesos 1.5 actually lands.

**Comment by jdef** [comment]
resolved by #341
